Odata-docs/webapi-8/fundamentals/metadata-routing.md

```diff
@@ -0,0 +1,235 @@
+    {
+        var modelBuilder = new ODataConventionModelBuilder();
+        modelBuilder.EntitySet<Customer>("Customers");
+        modelBuilder.EntitySet<Order>("Orders");
+
+        services.AddControllers().AddOData(
+            options => options.EnableQueryFeatures(null).AddRouteComponents(
+                routePrefix: "odata",
+                model: modelBuilder.GetEdmModel()));
+    }
+
+    public void Configure(IApplicationBuilder app)
+    {
+        app.UseODataRouteDebug();
+        app.UseRouting();
+        app.UseEndpoints(endpoints => endpoints.MapControllers());
+    }
+}
+```
+
+---
+
+In the preceding block of code, we define two entity sets, namely, `Customers` and `Orders`. Implicitly, `Customer` and `Order` get included in the Edm model as entity types.
+
+## Controller
+We do not need to add any controller(s) to the handle service metadata requests. Requests for service metadata are handled by a built-in controller named `MetadataController`. The controller exposes two actions, namely:
+- `GetServiceDocument` - generates the OData service document
+- `GetMetadata` - generates the OData service metadata document.
+
+## OData service document
+The service document is available from the root of the OData service. It gives you the titles and URLs for each of the OData service feeds.
+
+To fetch the service document, issue a GET request to the service root:
+```http
+GET http://localhost:5000/odata
+```
+
+The following JSON payload shows the expected response:
+```json
+{
+    "@odata.context": "http://localhost:5000/odata/$metadata",
+    "value": [
+        {
+            "name": "Customers",
+            "kind": "EntitySet",
+            "url": "Customers"
+        },
+        {
+            "name": "Orders",
+            "kind": "EntitySet",
+            "url": "Orders"
+        }
+    ]
+}
+```
+
+Based on the preceding payload, the entity sets in our OData service are `Customers` and `Orders`. The URLs for the two entity sets are `Customers` and `Orders` respectively.
+
+To retrieve all the customers, `Customers` would be appended to the service root to form the following URL:
+```http
+http://localhost:5000/odata/Customers
+```
+
+Similarly, to retrieve all the orders, `Orders` would be appended to the service root to form the following URL:
+```http
+http://localhost:5000/odata/Orders
+```
+
+## OData service metadata document
+The service metadata document contains a complete description of the feeds, types, properties, and relationships exposed by the OData service in the Entity Data Model (EDM).
+
+A service metadata document describes it's data in EDM terms using an XML language for describing models called the Conceptual Schema Definition Language (CSDL). When exposed by an OData service as a service metadata document, the CSDL document is packed using the EDMX format.
+
+Most OData clients libraries use the service metadata to drive the generation of client-side proxy classes to represent the server types.
```

Comment:
I wonder whether we should link to OData Connected Service and/or OData CLI here.